__tests__/Ship.test.js

```diff
@@ -0,0 +1,85 @@
+const { Ship, Port } = require("../src/ships.js");
+const dock = require("../src/ships.js");
+const Itinerary = require("../src/itinerary.js");
+
+describe("Ship", () => {
+  describe("Ship tests", () => {
+    let port;
+    let itinerary;
+    let ship;
+
+    beforeEach(() => {
+      port = new Port("Spithead");
+      /* port = {
+        addShip: jest.fn(),
+        removeShip: jest.fn(),
+        name: "Dover",
+        ships: [],
+      }; */
+      itinerary = new Itinerary([port]);
+      ship = new Ship(itinerary);
+    });
+
+    it("instantiates Ship object", () => {
+      expect(ship).toBeInstanceOf(Object);
+    });
+
+    it("gets added to port on instatiation", () => {
+      expect(port.ships).toContain(ship);
+    });
+
+    it("has a starting port", () => {
+      expect(ship.currentPort).toBe(port);
+    });
+  });
+});
+
+describe("Ship function tests", () => {
+  let spithead;
+  let botanyBay;
+  let itinerary;
+  let ship;
+
+  beforeEach(() => {
+    spithead = {
+      addShip: jest.fn(),
+      removeShip: jest.fn(),
+      name: "Dover",
+      ships: [],
+    };
+
+    botanyBay = {
+      addShip: jest.fn(),
+      removeShip: jest.fn(),
+      name: "Botany Bay",
+      ships: [],
+    };
+
+    itinerary = {
+      ports: [spithead, botanyBay],
+    };
+    ship = new Ship(itinerary);
+  });
+
+  it("can set sail", () => {
+    ship.setSail();
+
+    expect(ship.currentPort).toBeFalsy();
+    expect(spithead.removeShip).toHaveBeenCalledWith(ship);
+  });
+
+  it("can dock at a different port", () => {
+    ship.setSail();
+    ship.dock();
+
+    expect(ship.currentPort).toBe(botanyBay);
+    expect(botanyBay.ships).not.toContain(ship);
```

Comment:
because you're using stubs, `botanyBay` would never contain `ship` (the function `jest.fn()` isn't actually doing anything). Furthermore, you are expecting the `botanyBay` `addShip` method to be called (it's the second ship in itinerary), so this test is also misleading (if it wasn't a stub, `botanyBay.ships` would contain `ship`) Instead, I would change this expect statement to
```
expect(botanyBay.addShip).toHaveBeenCalledWith(ship)
```
(similar to how you've tested `spithead.removShip` in `can set sail` on line 68)